streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBWindowStoreTest.java

```diff
@@ -605,36 +1066,138 @@ public void testRestore() throws IOException {
+        assertEquals(
+            Collections.emptyList(),
+            toList(windowStore.fetch(
+                0,
+                ofEpochMilli(startTime - windowSize),
+                ofEpochMilli(startTime + windowSize))));
+        assertEquals(
+            Collections.emptyList(),
+            toList(windowStore.fetch(
+                1,
+                ofEpochMilli(startTime + increment - windowSize),
+                ofEpochMilli(startTime + increment + windowSize))));
+        assertEquals(
+            Collections.emptyList(),
+            toList(windowStore.fetch(
+                2,
+                ofEpochMilli(startTime + increment * 2 - windowSize),
+                ofEpochMilli(startTime + increment * 2 + windowSize))));
+        assertEquals(
+            Collections.emptyList(),
+            toList(windowStore.fetch(
+                3,
+                ofEpochMilli(startTime + increment * 3 - windowSize),
+                ofEpochMilli(startTime + increment * 3 + windowSize))));
+        assertEquals(
+            Collections.singletonList("four"),
+            toList(windowStore.fetch(
+                4,
+                ofEpochMilli(startTime + increment * 4 - windowSize),
+                ofEpochMilli(startTime + increment * 4 + windowSize))));
+        assertEquals(
+            Collections.singletonList("five"),
+            toList(windowStore.fetch(
+                5,
+                ofEpochMilli(startTime + increment * 5 - windowSize),
+                ofEpochMilli(startTime + increment * 5 + windowSize))));
+        assertEquals(
+            Collections.singletonList("six"),
+            toList(windowStore.fetch(
+                6,
+                ofEpochMilli(startTime + increment * 6 - windowSize),
+                ofEpochMilli(startTime + increment * 6 + windowSize))));
+        assertEquals(
+            Collections.singletonList("seven"),
+            toList(windowStore.fetch(
+                7,
+                ofEpochMilli(startTime + increment * 7 - windowSize),
+                ofEpochMilli(startTime + increment * 7 + windowSize))));
+        assertEquals(
+            Collections.singletonList("eight"),
+            toList(windowStore.fetch(
+                8,
+                ofEpochMilli(startTime + increment * 8 - windowSize),
+                ofEpochMilli(startTime + increment * 8 + windowSize))));
 
         // check segment directories
         windowStore.flush();
         assertEquals(
-            Utils.mkSet(segments.segmentName(4L), segments.segmentName(5L), segments.segmentName(6L)),
+            Utils.mkSet(
+                segments.segmentName(4L),
+                segments.segmentName(5L),
+                segments.segmentName(6L)),
             segmentDirs(baseDir)
         );
     }
 
+    private void verifyEmpty(final long startTime,
```

Comment:
Ditto on this method, which also appears to be used just once.

Reply:
compare comment above